edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/EdgeDeploymentStatusCommandTest.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test
+{
+    using System.Threading;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.ConfigSources;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+
+    using Xunit;
+
+    [Unit]
+    public class EdgeDeploymentStatusCommandTest
+    {
+        [Fact]
+        public async void ExecuteAyncSuccessWithNoStatus()
+        {
+            EdgeDeploymentStatusCommand noStatus = new EdgeDeploymentStatusCommand(Option.None<EdgeDeploymentStatus>());
+
+            await noStatus.ExecuteAsync(CancellationToken.None);
+        }
+
+        [Fact]
+        public async void ExecuteAyncSuccessWithSuccessStatus()
+        {
+            EdgeDeploymentStatus success = EdgeDeploymentStatus.Success("200 OK");
+            EdgeDeploymentStatusCommand successStatus = new EdgeDeploymentStatusCommand(Option.Some(success));
+
+            await successStatus.ExecuteAsync(CancellationToken.None);
+        }
+
+        [Fact]
+        public async void ExecuteAsyncThrowsWithFailureStatus()
+        {
+            EdgeDeploymentStatus failure = EdgeDeploymentStatus.Failure("failure message");
+            EdgeDeploymentStatusCommand failStatus = new EdgeDeploymentStatusCommand(Option.Some(failure));
+
+            await Assert.ThrowsAsync<ConfigOperationFailureException>(() => failStatus.ExecuteAsync(CancellationToken.None));
```

Comment:
How about the failure message? It should be ConfigOperationFailureException(failure.message), right?
While we eat the exception, are we able to collect the failure message and the inner exception?

Reply:
Can you confirm if the failure message will be eaten here or not?

Reply:
ping...